crypto/s2n_hkdf.c

```diff
@@ -46,7 +46,6 @@ static int s2n_custom_hkdf_extract(struct s2n_hmac_state *hmac, s2n_hmac_algorit
 {
     uint8_t hmac_size = 0;
     POSIX_GUARD(s2n_hmac_digest_size(alg, &hmac_size));
-    POSIX_ENSURE_LTE(hmac_size, pseudo_rand_key->size);
     pseudo_rand_key->size = hmac_size;
```

Comment:
You need to keep this safety check. You're changing the size of a blob, the new size better be <= the old size.

You can't assume that hmac_size < pseudo_rand_key->size because a higher function checked. The blob becomes invalid if you set the size wrong.